internal/matrix/mautrix_client.go

```diff
@@ -0,0 +1,71 @@
+package matrix
+
+import (
+	msg "neurobot/domain/message"
+	"neurobot/domain/room"
+
+	"maunium.net/go/mautrix"
+	"maunium.net/go/mautrix/event"
+	"maunium.net/go/mautrix/format"
+	mautrixId "maunium.net/go/mautrix/id"
+)
+
+type mautrixClient interface {
+	Login(*mautrix.ReqLogin) (*mautrix.RespLogin, error)
+	SendText(roomID mautrixId.RoomID, text string) (*mautrix.RespSendEvent, error)
+	SendMessageEvent(roomID mautrixId.RoomID, eventType event.Type, contentJSON interface{}, extra ...mautrix.ReqSendEvent) (resp *mautrix.RespSendEvent, err error)
+	ResolveAlias(alias mautrixId.RoomAlias) (resp *mautrix.RespAliasResolve, err error)
+}
+
+type client struct {
```

Comment:
@psrpinto Am I right to assume this would eventually be an implementation of a modified MatrixClient interface?